### PR TITLE
For #968. Removing Guava's code.

### PR DESCRIPTION
**Pull request by zCRUSADERz**
For #968
- Updated cactoos-matchers to 0.15
- Replaced Guava's code on cactoos
- Replaced static matchers on cactoos-matchers
- Updated puzzle

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#982](https://github.com/yegor256/takes/issues/982) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#982](https://github.com/yegor256/takes/issues/982) is assigned to @serranya[/z](https://www.0crat.com/u/serranya), here is [why](https://www.0crat.com/footprint/C3FFK3YAY/118f521d-03ba-4dbb-9bd9-a6bf3eb5020a); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: .travis.yml [hunk 1]
```diff
@@ -11,8 +11,8 @@ script:
   - est --dir=est --file=/dev/null
   - mvn clean install -Pqulice -Pjacoco --errors --batch-mode
 before_install:
-  - rvm install 2.2.2
-  - rvm use 2.2.2
+  - rvm install 2.6.0
```
Comment (Serranya) [inline .travis.yml]: @zCRUSADERz  Why did you bump the version? This was not part of the task.

Comment (zCRUSADERz) [reply .travis.yml]: @Serranya I needed a successful build in Travis CI. Probably I should wait until #981 PR is accepted?

Comment (Serranya) [reply .travis.yml]: @zCRUSADERz exactly. Please remove this part of the commit and wait for #981

Comment (zCRUSADERz) [reply .travis.yml]: @Serranya deleted.

**Comment by Serranya** [comment]
@zCRUSADERz There are test failures. Please take a look

**Comment by zCRUSADERz** [comment]
@Serranya on my machine, all tests passed, and AppVeyor build succeeded. Travis CI failed, because pdd gem requires ruby >= 2.3.0. See #980, #981

**Comment by zCRUSADERz** [comment]
@Serranya all checks have passed.

**Comment by Serranya** [comment]
@paulodamaso looking good

**Comment by paulodamaso** [comment]
@rultor merge

**Comment by 0crat** [comment]
Code review was too long (8 days), architects (@paulodamaso) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @serranya[/z](https://www.0crat.com/u/serranya), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#982](https://github.com/yegor256/takes/issues/982) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by sereshqua** [comment]
@Serranya please make sure that next time you will try to find at least 3 issues (if possible) during the CR

**Comment by sereshqua** [comment]
@Serranya ping

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @serranya[/z](https://www.0crat.com/u/serranya)

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)

**Pull request merged**